src/Forms/GridField/GridField.php

```diff
@@ -1435,4 +1438,20 @@ private function addSudoModeComponent(): void
         $newConfig->addComponents($components);
         $this->setConfig($newConfig);
     }
+
+    /**
+     * Prepopulate the version number cache if the list is a DataList
+     * and the DataObject class it contains has the Versioned extension
+     */
+    private function prepopulateVersionNumberCache(SS_List $list): void
+    {
+        if (is_a($list, DataList::class)
+            && class_exists(Versioned::class)
+            && $list->dataClass()::has_extension(Versioned::class)
```

Comment:
Do we even need the `has_extension` check here? Would it make more sense to just call `$list->prepopulateCaches()` in `FieldHolder()` regardless of whether the base class is versioned? The versioned `onPrepopulateCaches()` hook checks `$dataClass::has_extension(Versioned::class)` already.

In fact we don't need the conflict either now that I think about it - because on versioned `3.0` it just won't have the extension hook anyway.

Reply:
Updated